include/wx/generic/dirctrlg.h

```diff
@@ -269,6 +314,8 @@ class WXDLLIMPEXP_CORE wxDirFilterListCtrl: public wxChoice
 // Symbols for accessing individual controls
 #define wxID_TREECTRL          7000
 #define wxID_FILTERLISTCTRL    7001
+#define wxID_MENU_DIR          7002
```

Comment:
What meaning do these constants have? Menus don't have IDs... only menu items do.

Reply:
They are IDs for the showing menu events, to distinguish between a File menu and a Directory menu event.  Perhaps they should have different names?  E.G. `wxCommandEvent event2(wxEVT_DIRCTRL_SHOWING_POPUP_MENU, wxID_MENU_DIR)`

Reply:
Event ID must correspond to some object (control, menu item, ...) with this ID. So it definitely doesn't make sense to use them to distinguish between the events of different types.

I don't think we need multiple events neither, we do have the item for which the menu is shown and hence its `wxDirItemData` and so its `m_isDir`.